ibm/service/kubernetes/resource_ibm_container_cluster.go

```diff
@@ -701,26 +701,15 @@ func resourceIBMContainerClusterCreate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
-	_, err = waitForClusterMasterAvailable(d, meta)
+	_, err = waitForClusterMasterAvailable(d, meta, d.Timeout(schema.TimeoutCreate))
 	if err != nil {
 		return err
 	}
-	waitForState := strings.ToLower(d.Get("wait_till").(string))
-
-	switch waitForState {
-	case strings.ToLower(oneWorkerNodeReady):
-		_, err = waitForClusterOneWorkerAvailable(d, meta)
-		if err != nil {
-			return err
-		}
-
-	case strings.ToLower(clusterNormal):
-		pendingStates := []string{clusterDeploying, clusterRequested, clusterPending, clusterDeployed, clusterCritical, clusterWarning}
-		_, err = waitForClusterState(d, meta, waitForState, pendingStates)
-		if err != nil {
-			return err
-		}
 
+	timeoutStage := strings.ToLower(d.Get("wait_till").(string))
```

Comment:
The default value of `wait_till` is IngressReady . If user's don't provide attribute by default the provisioning waits till ingressReady need to handle that scenario in waitForCluster